src/Data/LargeHashable/Intern.hs

```diff
@@ -6,67 +6,145 @@
 --
 -- Regular users should not import this module. Import 'Data.LargeHashable'
 -- instead.
+{-# LANGUAGE BangPatterns #-}
+{-# LANGUAGE DeriveGeneric #-}
 {-# LANGUAGE GeneralizedNewtypeDeriving #-}
 module Data.LargeHashable.Intern (
 
-    HashUpdates(..), HashAlgorithm(..), LH, Hash(..)
-  , hashUpdates, ioInLH, runLH
+    HashUpdates(..), HashAlgorithm(..), LH
+  , hashUpdates, ioInLH, runLH, updateXorHash
 
 ) where
 
-import Control.Monad.Trans.Reader
-import Control.Monad.IO.Class
+-- keep imports in alphabetic order (in Emacs, use "M-x sort-lines")
+import Control.Monad
 import Data.Word
-import Foreign.Ptr
 import Foreign.C.Types
+import Foreign.Ptr
 import System.IO.Unsafe (unsafePerformIO)
-import qualified Data.ByteString as BS
-import qualified Data.ByteString.Char8 as BSC
-import qualified Data.ByteString.Base16 as Base16
-
--- | A newtype-wrapper for hash values.
-newtype Hash = Hash { unHash :: BS.ByteString }
-    deriving (Eq)
-
-instance Show Hash where
-    show (Hash bs) =
-        BSC.unpack (Base16.encode bs)
 
 -- | Functions for updating an intermediate hash value. The functions live
 -- in the 'IO' monad because they are typically implemented via FFI.
--- FIXME: functions for primitive types other than CULong missing.
 data HashUpdates
     = HashUpdates
```

Comment:
No, the could be lazy. But I feared that someone makes them strict, without notifying the performance regression caused by unbox-strict-fields.